### DISCUSSION TITLE
Sensible defaults for available entrance styles

### DISCUSSION THREAD

**Discussion by zzril** [OpenRCT2/OpenRCT2#21224] [Feature requests]
The availability of OpenRCT2's extra entrance styles ("no entrance" and "no platform / no entrance") in scenario play and track designer seems a bit random to me.

Using only RCT 2 as base game, these styles are available by default:  
* in sc6 scenarios from the base game: all 12 base game styles + "no entrance"
* in track designer: all 12 base game styles
* in scenario editor: all 12 base game styles are selected by default, "no entrance" and "no platform / no entrance" are unselected by default
* in newly created scenarios: entrance styles are available as defined when creating the scenario

I also noticed that it is possible to create scenarios that do not have the "plain" entrance style available, i. e. that one is not marked as required.

---

For the old sc6 scenarios, having "no entrance" available, but not "no platform / no entrance" seems kind of odd to me.

For the track designer, I think that definitely all entrance styles should be available. (I. e. also the 2 new entryless / platformless ones.)

Concerning the "plain" entrance style, I think that it should be always required in new scenarios. (Mainly because otherwise the decision what entrance style to pre-select for a new, custom design in a fresh park without any rides would be a bit random.)

Would like to hear others' opinions on this.

**Comment by Gymnasiast** [comment]
Seems quite sensible to me for the most part. I would argue that the "no entrance" and "no entrance, no platform" options should not be selected by default in any circumstance, as they look a bit weird without cheats.

The most important reason why "no entrance" is available in SV6/SC6 is just historical: we hacked that in when we were still using SV6. This should be modified but requires some code to determine whether the "no entrance" style is used or not.
